pettingzoo/utils/conversions.py

```diff
@@ -305,11 +305,14 @@ def reset(self, seed=None, options=None):
         self.truncations = {agent: False for agent in self.agents}
         self.rewards = {agent: 0 for agent in self.agents}
 
-        # Every environment needs to return infos that contain the same keys as the observations
-        if bool(self.infos) == False:
+        # Every environment needs to return infos that contain self.agents as their keys
+        if not self.infos:
             warnings.warn("The `infos` dictionary returned by `env.reset` was empty. Agent IDs as keys will be used")
             self.infos = {agent: {} for agent in self.agents}
-
+        elif set(self.infos.keys()) != set(self.agents):
+            warnings.warn("The `infos` dictionary keys returned by `env.reset` is not equal to the agent IDs listed in `self.agents`. Overwriting with just agent IDs as keys")
```

Comment:
I think the warning should be more like the infos returned by env.reset is not valid: must contain keys for each agent defined in `self.agents`: {self.agents}. Duplicating current info data to each agent... {self.infos}"